OPHD/States/PlanetSelectState.cpp

```diff
@@ -192,44 +194,11 @@ State* PlanetSelectState::update()
 	{
 		return this;
 	}
-	else if (PLANET_TYPE_SELECTION != Planet::PlanetType::None)
+	else if (planetSelection != planetSelectionInvalid)
 	{
-		std::string map, tileset;
-		int dig_depth = 0, max_mines = 0;
-		Planet::Hostility hostility = Planet::Hostility::None;
+		const Planet& planet = planetAttributes[planetSelection];
 
-		switch (PLANET_TYPE_SELECTION)
-		{
-		case Planet::PlanetType::Mercury:
-			map = "maps/merc_01";
-			tileset = "tsets/mercury.png";
-			dig_depth = mPlanets[0]->digDepth();
-			max_mines = mPlanets[0]->maxMines();
-			hostility = Planet::Hostility::High;
-			break;
-
-		case Planet::PlanetType::Mars:
-			map = "maps/mars_04";
-			tileset = "tsets/mars.png";
-			dig_depth = mPlanets[1]->digDepth();
-			max_mines = mPlanets[1]->maxMines();
-			hostility = Planet::Hostility::Low;
-			break;
-
-		case Planet::PlanetType::Ganymede:
-			map = "maps/ganymede_01";
-			tileset = "tsets/ganymede.png";
-			dig_depth = mPlanets[2]->digDepth();
-			max_mines = mPlanets[2]->maxMines();
-			hostility = Planet::Hostility::Medium;
-			break;
-
-		default:
-			return mReturnState;
-			break;
-		}
-
-		MapViewState* mapview = new MapViewState(map, tileset, dig_depth, max_mines, hostility);
+		MapViewState* mapview = new MapViewState(planet.mapImagePath(), planet.tilesetPath(), planet.digDepth(), planet.maxMines(), planet.hostility());
```

Comment:
I am thinking about passing planet into MapViewState instead of all the parameters.

I think Planet should be separated into a Planet and a PlanetView class. PlanetView would be responsible for drawing the planet instead of Planet handling its own draw logic.

Reply:
Makes sense. I think you're on to something here with this refactor.

When I was reading the changes, I was kind of thinking rather than store individual fields and copy fields one at a time, you could contain the struct and copy the struct as a whole. Could do similar with the accessors, replacing accessors for each attribute with a single accessor to get a `const &` reference to the struct. Struct fields are public, so the caller would be free to read whatever subfields from there, without have to write explicit accessors for all of them.